Sparkle/SUBinaryDeltaApply.m

```diff
@@ -15,38 +15,84 @@
 #include <stdlib.h>
 #include <xar/xar.h>
 
-static void applyBinaryDeltaToFile(xar_t x, xar_file_t file, NSString *sourceFilePath, NSString *destinationFilePath)
+static BOOL applyBinaryDeltaToFile(xar_t x, xar_file_t file, NSString *sourceFilePath, NSString *destinationFilePath)
 {
     NSString *patchFile = temporaryFilename(@"apply-binary-delta");
     xar_extract_tofile(x, file, [patchFile fileSystemRepresentation]);
     const char *argv[] = {"/usr/bin/bspatch", [sourceFilePath fileSystemRepresentation], [destinationFilePath fileSystemRepresentation], [patchFile fileSystemRepresentation]};
-    bspatch(4, (char **)argv);
+    BOOL success = (bspatch(4, (char **)argv) == 0);
     unlink([patchFile fileSystemRepresentation]);
+    return success;
 }
 
 int applyBinaryDelta(NSString *source, NSString *destination, NSString *patchFile)
 {
-    xar_t x = xar_open([patchFile UTF8String], READ);
+    xar_t x = xar_open([patchFile fileSystemRepresentation], READ);
     if (!x) {
-        fprintf(stderr, "Unable to open %s. Giving up.\n", [patchFile UTF8String]);
+        fprintf(stderr, "Unable to open %s. Giving up.\n", [patchFile fileSystemRepresentation]);
         return 1;
     }
+    
+    SUBinaryDeltaMajorVersion majorDiffVersion = FIRST_DELTA_DIFF_MAJOR_VERSION;
+    SUBinaryDeltaMinorVersion minorDiffVersion = FIRST_DELTA_DIFF_MINOR_VERSION;
```

Comment:
They might not be necessary, but it may be nice info to have. They can actually affect behavior though. For example, before this PR I made a couple backwards compatible changes for creating the diffs: case sensitive file renaming issue and a failing case where a directory became a symbolic link. For these cases, the minor version would be bumped (and in this patch I bumped it to 1.1 when creating version 1 patches). If I sent a version 1 patch to someone to apply, they can tell if the patch was created by version 1.0 or 1.1, which may be the difference for the patch applying correctly.

Thoughts?